website/api/controllers/try-fleet/view-query-report.js

```diff
@@ -0,0 +1,237 @@
+    if(hostPlatform === 'macos'){
+      osqueryTablesToDisplay = _.filter(sails.config.builtStaticContent.osqueryTables, (table)=>{
+        return _.contains(table.platforms, 'darwin') && queryIdsByTableName[`${table.name}`] !== undefined;
+      });
+    } else if(hostPlatform === 'linux'){
+      osqueryTablesToDisplay = _.filter(sails.config.builtStaticContent.osqueryTables, (table)=>{
+        return _.contains(table.platforms, 'linux') && queryIdsByTableName[`${table.name}`] !== undefined;
+      });
+    } else if(hostPlatform === 'windows'){
+      osqueryTablesToDisplay = _.filter(sails.config.builtStaticContent.osqueryTables, (table)=>{
+        return _.contains(table.platforms, 'windows') && queryIdsByTableName[`${table.name}`] !== undefined;
+      });
+    }
+
+    // If the specified table does not exist, or is not compatible with the selected host
+    if(!_.contains(_.pluck(osqueryTablesToDisplay, 'name'), tableName)){
+      throw 'invalidTable';
+    }
+    let specifiedOsqueryTable = _.find(osqueryTablesToDisplay, {'name': tableName});
+
+    //  ┌─┐ ┬ ┬┌─┐┬─┐┬ ┬  ┬─┐┌─┐┌─┐┬ ┬┬ ┌┬┐┌─┐
+    //  │─┼┐│ │├┤ ├┬┘└┬┘  ├┬┘├┤ └─┐│ ││  │ └─┐
+    //  └─┘└└─┘└─┘┴└─ ┴   ┴└─└─┘└─┘└─┘┴─┘┴ └─┘
+
+    let queryIdToGetReportFor = queryIdsByTableName[`${tableName}`];
+    // Send an HTTP request to get the query report for the query for this table.
+    let queryReportResponse = await sails.helpers.http.get.with({
+      url: sails.config.custom.fleetBaseUrlForQueryReports+'/api/v1/fleet/queries/'+queryIdToGetReportFor+'/report',
+      headers: {
+        Authorization: `Bearer ${sails.config.custom.fleetTokenForQueryReports}`
+      }
+    })
+    .intercept((error)=>{
+      return new Error(`When sending an API request to ${sails.config.custom.fleetBaseUrlForQueryReports}/api/v1/fleet/queries/${queryIdToGetReportFor}/report to get the latest query report for the ${tableName} table, an error occured: ${error}`);
+    });
+
+    let queryResults = queryReportResponse.results;
+    // Group the query results by the host that reported them.
+    let queryResultsByHostIds = _.groupBy(queryResults, 'host_id');
+    // Default these to empty arrays, if there are no results for this host, we'll send an empty array of results array to the page and show the user an empty state.
+    let reportForThisHost = [];
+    let reportWithSortedColumns = [];
+    let topResultLastFetchedAt = 0;
+
+    // Process the query results for this host (If there are any)
+    if(queryResultsByHostIds[selectedHostId]) {
+      let resultsForThisHost = queryResultsByHostIds[selectedHostId];
+      // Sort the results by their last fetched value.
+      let resultsOrderedByLastFetched = _.sortByOrder(resultsForThisHost, 'last_fetched');
+      // Get a timestamp of when the last result was fetched from the host.
+      topResultLastFetchedAt = Date.parse(resultsOrderedByLastFetched[0].last_fetched);
```

Comment:
worth glancing at this timestamp and making sure what we see matches what's in dogfood and it's not getting greenwitched